modules/ting_search_carousel/ting_search_carousel.pages.inc

```diff
@@ -20,10 +20,13 @@ function ting_search_carousel_result($index, $offset) {
   }
   $offset = $chunk['next_offset'];
 
+  $js = drupal_add_js();
+
   // Return JSON output.
   drupal_json_output(array(
     'content' => $content,
     'index' => $index,
     'offset' => $offset,
+    'settings' => drupal_array_merge_deep_array($js['settings']['data']),
```

Comment:
I am not a fan of this approach. It pollutes the concept of settings in the behavior. I wonder why it is needed. That is hard to tell from the pull request and commit comments.

If the rendering of the carousel included data that the frontend code needs to react properly then I would much prefer that this was contained with the rendering and not spread between rendering and settings. Would it be possible to add the required data as a data attribute instead?

Reply:
In this specific issue, we needed to include the settings, so the ding_availability settings would be included. Adding that information to a `data-` attribute would require a rewrite of how ding_availability works.
Including all the settings should allow other modules that also utilize the settings array to work on the materials loaded in the carousel.